src/NBitcoin/Network.cs

```diff
@@ -230,6 +230,102 @@ internal static Network Register(Network network)
             return network;
         }
 
+        /// <summary>
+        /// Mines a new genesis block, to use with a new network.
+        /// Typically, 3 such genesis blocks need to be created when bootstrapping a network: for Main, Test and Reg.
+        /// </summary>
+        /// <param name="consensusFactory">
+        /// The consensus factory used to create transactions and blocks. 
+        /// Use <see cref="PosConsensusFactory"/> for proof-of-stake based networks.
+        /// </param>
+        /// <param name="pszTimestamp">
+        /// Traditionally a news headline from the day of the launch, but could be any string or link.
+        /// This will be inserted in the input coinbase transaction script.
+        /// </param>
+        /// <param name="target">
+        /// The difficulty target under which the hash of the block need to be. 
+        /// Make the Test and Reg targets ones easier to find than the Main one so that you don't wait too long to mine the genesis block.
+        /// </param>
+        /// <param name="genesisReward">
+        /// Specify how many coins to put in the genesis transaction's output. These coins are unspendable.
```

Comment:
Not sure if this value should be exposed. Why not just default it to 50 coins or smth? 

Btw I didn't know about the genesis reward and why it's unspendable: 

quote:
This is how the Bitcoin client worked when Satoshi created it: The client maintains a block database and a transaction database. When it finds that its block database is empty, it inserts the genesis block into its block database to get things started. This block includes a transaction sending 50 BTC to 1A1z..., but the client does not insert this transaction into its transaction database. So even though this transaction is part of the block chain, if the client sees a transaction spending this 50 BTC, it won't be able to find the 50 BTC transaction in its transaction database, and the spending transaction will be rejected. In other words, the genesis block's transaction isn't considered to be a "real transaction" by the original Bitcoin client.

(Current versions of Bitcoin-Qt handle the block/transaction databases in a totally different way, so the genesis block's transaction is now just a weird special case in the code.)

This could easily have been an oversight, or Satoshi could have intended this transaction to be unspendable. This quirk wasn't discovered until after Satoshi disappeared, so no one had a chance to ask him.

from https://www.reddit.com/r/Bitcoin/comments/1nc13r/the_first_50btc_block_reward_cant_be_spend_why/

Reply:
ahah yeah that's interesting.
I guess that's one of these things that has now become "tradition".

Reply:
For the 50 coins, we just let the user decide. Doesn't really matter either way. 
In Stratis Test and Reg we set it to 0 for example.

Reply:
Makes sense. 
Just wanted to avoid users thinking on what value to use when setting up a new network. 
But since comment says that it's fine.